File: frontend/projects/api/src/lib/events/list-my-events-response.ts
import { EventSummary } from './event-summary';

export interface ListMyEventsResponse {
  readonly thisWeek: ReadonlyArray<EventSummary>;
  readonly laterThisMonth: ReadonlyArray<EventSummary>;
  readonly nextMonth: ReadonlyArray<EventSummary>;
  readonly past: ReadonlyArray<EventSummary>;
}

File: backend/src/PullUp.Application/Features/Events/ListMyEvents/ListMyEventsResponse.cs
namespace PullUp.Application.Features.Events.ListMyEvents;

public sealed record ListMyEventsResponse(
    IReadOnlyList<EventSummary> ThisWeek,
    IReadOnlyList<EventSummary> LaterThisMonth,
    IReadOnlyList<EventSummary> NextMonth,
    IReadOnlyList<EventSummary> Past);
